src/query/api/v1/handler/prometheus/common.go

```diff
@@ -152,6 +155,61 @@ func parseTagCompletionQuery(r *http.Request) (string, error) {
 	return queries[0], nil
 }
 
+func parseTimeWithDefault(
+	r *http.Request,
+	key string,
+	defaultTime time.Time,
+) (time.Time, error) {
+	if t := r.FormValue(key); t != "" {
+		return util.ParseTimeString(t)
+	}
+
+	return defaultTime, nil
+}
+
+// ParseSeriesMatchQuery parses all params from the GET request
+func ParseSeriesMatchQuery(
+	r *http.Request,
+	tagOptions models.TagOptions,
+) (*storage.SeriesMatchQuery, *xhttp.ParseError) {
+	r.ParseForm()
+	matcherValues := r.Form["match[]"]
+	if len(matcherValues) == 0 {
+		return nil, xhttp.NewParseError(errors.ErrInvalidMatchers, http.StatusBadRequest)
+	}
+
+	start, err := parseTimeWithDefault(r, "start", time.Now().Add(time.Hour*24*-40))
```

Comment:
move the default to a constant ?

Reply:
It uses time.Now, should not be a constant